tutorials/simulation/10_array_objs.py

```diff
@@ -195,3 +195,49 @@
                                nave=data.shape[0], comment='simulated')
 print(evoked_array)
 evoked_array.plot()
+
+
+# %%
+# In certain situations you may wish to use a custom time-frequency
+# decomposition for estimation of power spectra. Or you may wish to
+# process pre-computed power spectra in MNE.
+# Following the same logic, it is possible to instantiate averaged power
+# spectrum using the :class:`~mne.time_frequency.Spectrum` class.
+# This is slightly
+# experimental at the moment but works. An API for doing this may follow.
+
+# compute power spectrum
+
+psd, freqs = mne.time_frequency.psd_array_welch(
+    data, info['sfreq'], n_fft=128, n_per_seg=32)
+
+psd_ave = psd.mean(0)
+
+# map to `~mne.time_frequency.Spectrum` class and explore API
+
+
+def spectrum_from_array(data: np.ndarray,  # spectral features
+                        freqs: np.ndarray,  # frequencies
+                        inst_info: mne.Info  # the meta data of MNE instance
+                        ) -> mne.time_frequency.Spectrum:  # Spectrum object
+    """Create MNE averaged power spectrum object from custom data"""
+    state = dict(
+        method='my_welch',
+        data=data,
+        sfreq=inst_info['sfreq'],
+        dims=('channel', 'freq'),
+        freqs=freqs,
+        inst_type_str='Raw',
+        data_type='simulated signals',
```

Comment:
@drammock I put this for illustration purposes. Are there any expectations in the Spectrum code regarding this field? It was 'Averaged EEG' in my concrete case on which this example is based.

Reply:
right now it is only used in the `repr` of the object. I don't have any concrete plans to use it in any other way (and I would actively *avoid* using it in `if` clauses to triage different flavors of spectrum, it's probably not reliable to do so)

Reply:
native values for this are `Fourier Coefficients`, `Power Spectrum`, and `Averaged Power Spectrum`. `Averaged Fourier Coefficients` is not supported, see 

https://github.com/mne-tools/mne-python/blob/18474b1d9227778ae875692a56f7e0927e856a54/mne/time_frequency/spectrum.py#L1109-L1118